cai_causal_graph/time_series_causal_graph.py

```diff
@@ -681,7 +673,43 @@ def add_edge(
                 )
         # No else needed as we don't need to check time direction for other edge types.
 
-        edge = super().add_edge(source, destination, edge_type=edge_type, meta=meta, edge=edge, **kwargs)
+    @_reset_ts_graph_attributes
+    def add_edge(
+        self,
+        /,
+        source: Optional[NodeLike] = None,
+        destination: Optional[NodeLike] = None,
+        *,
+        edge_type: EdgeType = EdgeType.DIRECTED_EDGE,
+        meta: Optional[dict] = None,
+        edge: Optional[Edge] = None,
+        **kwargs,
+    ) -> Edge:
+        """
+        Add an edge to the graph. See `cai_causal_graph.causal_graph.CausalGraph.add_edge` for more details.
+
+        In addition to the `cai_causal_graph.causal_graph.CausalGraph.add_edge` method, this method also populates the
+        metadata of the nodes with the variable name and the time lag.
+
+        :param source: The source node of the edge.
+        :param destination: The destination node of the edge.
+        :param edge_type: The type of edge to add.
+        :param meta: The metadata to add to the edge.
+        :param edge: The edge to add.
+        :param kwargs: Additional keyword arguments to pass to the
+            `cai_causal_graph.causal_graph.TimeSeriesCausalGraph.add_edge` method.
+        :return: The edge that was added.
+        """
+        # if the nodes are Node but not TimeSeriesNodes, covert them to TimeSeriesNodes
+        if edge is not None and isinstance(edge.source, Node) and not isinstance(edge.source, TimeSeriesNode):
+            new_source = TimeSeriesNode.from_dict(edge.source.to_dict())
+            new_destination = TimeSeriesNode.from_dict(edge.destination.to_dict())
+            edge = Edge(new_source, new_destination, edge_type=edge_type, meta=meta, **kwargs)
```

Comment:
This is wrong as you are overwriting the edge type and meta that was on the original `edge`. If `edge` is provided we ignore all other parameters. See the docstring for `add_edge` in the base class. But this is moot as I provided updated code above.